core/src/main/java/org/infinispan/Version.java

```diff
@@ -24,25 +26,39 @@
 
    private static final Version INSTANCE = new Version();
 
+   private final String version;
+   private final String codename;
    private final byte[] versionId;
    private final String moduleSlot;
    private final short versionShort;
    private final short marshallVersion;
    private final String majorMinor;
    private final String major;
 
+
    private Version() {
-      String parts[] = getParts(Injected.getVersion());
+      Properties properties = new Properties();
+      try (InputStream is = Version.class.getResourceAsStream("/META-INF/infinispan-version.properties")) {
```

Comment:
BTW this is my approach using standard JDK API:
 - https://github.com/hibernate/hibernate-ogm/pull/708/commits/dc4cfeade80e16520f04d4db3f34461d0255fd82

Doesn't get you all the strange formats you want though ;) Maybe get rid of some?